### PR TITLE
Generalize the wild-card fqdn's for user_agent attributes in ssl_server_name

### PR DESCRIPTION
**Pull request by shinrich**
Add wildcard ssl_server_name tests for each of the features: verify_client, tunnel_route, h2disable, verify_origin, client_cert.

Make fixes to ensure that the basic glob style wildcards work for all features.  Lay groundwork for future additional of full regex fqdn matching.

### CODE REVIEW

File: iocore/net/SSLNetVConnection.cc [hunk 1]
```diff
@@ -973,8 +975,7 @@ SSLNetVConnection::sslStartHandShake(int event, int &err)
           return EVENT_DONE;
         } else {
           SSLConfig::scoped_config params;
```
Comment (SolidWallOfCode) [inline iocore/net/SSLNetVConnection.cc]: What is the point of `params`?

File: iocore/net/SSLSNIConfig.cc [hunk 2]
```diff
@@ -35,55 +35,49 @@
...
   }
   return nps;
 }
 
 void
 SNIConfigParams::loadSNIConfig()
 {
-  for (const auto &item : Y_sni.items) {
-    actionVector *aiVec = new actionVector();
+  for (auto item : Y_sni.items) {
```
Comment (SolidWallOfCode) [inline iocore/net/SSLSNIConfig.cc]: Is there a reason to change this to make copies of the items?

File: iocore/net/SSLNetVConnection.cc (hunk 1 above)
Comment (shinrich) [reply iocore/net/SSLNetVConnection.cc]: Yes, no longer needed since sni_map_enable is removed.

**Comment by shinrich** [comment]
Addressed Alan's comments and rebased against @chitianhao's changes.

File: iocore/net/P_SSLSNI.h [hunk 3]
```diff
@@ -41,32 +41,77 @@
...
+  setGlobName(std::string &name)
+  {
+    std::string::size_type pos = 0;
+    while ((pos = name.find(".", pos)) != std::string::npos) {
+      name.replace(pos, 1, "\\.");
+      pos += 2;
+    }
+    pos = 0;
+    while ((pos = name.find("*", pos)) != std::string::npos) {
+      name.replace(pos, 1, ".{0,}");
```
[the remark this answers is not in the document]
Comment (SolidWallOfCode) [reply iocore/net/P_SSLSNI.h]: Why change it then?

**Comment by shinrich** [comment]
Pushed new commit to address Alan's last comment and bring in the wildcard tests in the recently merged h2disable tests.

**Pull request merged**